mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,394 @@
+		assumeRole.PolicyARNs = policiesARNs
+	}
+
+	assumeRole.RoleARN = tfAssumeRoleModel.RoleARN.ValueString()
+	assumeRole.SessionName = tfAssumeRoleModel.SessionName.ValueString()
+	assumeRole.SourceIdentity = tfAssumeRoleModel.SourceIdentity.ValueString()
+
+	if !tfAssumeRoleModel.TransitiveTagKeys.IsNull() {
+		var transitiveTagKeys []string
+		tfAssumeRoleModel.TransitiveTagKeys.ElementsAs(ctx, &transitiveTagKeys, true)
+		assumeRole.TransitiveTagKeys = transitiveTagKeys
+	}
+
+	return &assumeRole
+}
+
+func defaultValuesAndValidations(data *tfMongodbAtlasProviderModel, resp *provider.ConfigureResponse) {
+	if mongodbgovCloud := data.IsMongodbGovCloud.ValueBool(); mongodbgovCloud {
+		data.BaseURL = types.StringValue("https://cloud.mongodbgov.com")
+	}
+	if data.BaseURL.ValueString() == "" {
+		data.BaseURL = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_ATLAS_BASE_URL",
+			"MCLI_OPS_MANAGER_URL",
+		}, "").(string))
+	}
+
+	if data.PublicKey.ValueString() == "" {
+		data.PublicKey = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_ATLAS_PUBLIC_KEY",
+			"MCLI_PUBLIC_API_KEY",
+		}, "").(string))
+		if data.PublicKey.ValueString() == "" {
+			resp.Diagnostics.AddError(ProviderConfigError, fmt.Sprintf(AttrNotSetError, "public_key"))
+		}
+	}
+
+	if data.PrivateKey.ValueString() == "" {
+		data.PrivateKey = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_ATLAS_PRIVATE_KEY",
+			"MCLI_PRIVATE_API_KEY",
+		}, "").(string))
+		if data.PrivateKey.ValueString() == "" {
+			resp.Diagnostics.AddError(ProviderConfigError, fmt.Sprintf(AttrNotSetError, "private_key"))
+		}
+	}
+
+	if data.RealmBaseURL.ValueString() == "" {
+		data.RealmBaseURL = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_REALM_BASE_URL",
+		}, "").(string))
+	}
+
+	if data.Region.ValueString() == "" {
+		data.Region = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_REGION",
+			"TF_VAR_AWS_REGION",
+		}, "").(string))
+	}
+
+	if data.StsEndpoint.ValueString() == "" {
+		data.StsEndpoint = types.StringValue(MultiEnvDefaultFunc([]string{
+			"STS_ENDPOINT",
+			"TF_VAR_STS_ENDPOINT",
+		}, "").(string))
+	}
+
+	if data.AwsAccessKeyID.ValueString() == "" {
+		data.AwsAccessKeyID = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_ACCESS_KEY_ID",
+			"TF_VAR_AWS_ACCESS_KEY_ID",
+		}, "").(string))
+	}
+
+	if data.AwsSecretAccessKeyID.ValueString() == "" {
+		data.AwsSecretAccessKeyID = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_SECRET_ACCESS_KEY",
+			"TF_VAR_AWS_SECRET_ACCESS_KEY",
+		}, "").(string))
+	}
+
+	if data.AwsSessionToken.ValueString() == "" {
+		data.AwsSessionToken = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_SESSION_TOKEN",
+			"TF_VAR_AWS_SESSION_TOKEN",
+		}, "").(string))
+	}
+}
+
+func (p *MongodbtlasProvider) DataSources(context.Context) []func() datasource.DataSource {
+	return []func() datasource.DataSource{
+		// NewExampleDataSource,
```

Comment:
```suggestion
```

Reply:
Is being deleted in project migration PR (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1365/files#diff-50a1299c66e854062730103c8144024ebf3a8ecc6f1b620231d15ed6f5997424L354)